packages/components/src/components/ResizableBox/ResizableBox.tsx

```diff
@@ -0,0 +1,209 @@
+import styled from 'styled-components';
+import { useEffect, useRef, useState } from 'react';
+
+export type ResizableBoxProps = {
+    children: React.ReactNode;
+    directions: Array<"top" | "left" | "right" | "bottom">; 
+    isLocked?: boolean; // if true, it's not possible to resize the component
+    width?: number;
+    minWidth?: number;
+    maxWidth?: number;
+    height?: number;
+    minHeight?: number;
+    maxHeight?: number;
+};
+
+const reactiveAreaWidth = '15px';
+
+type ResizersProps = {
+    $width?: number;
+    $minWidth?: number;
+    $maxWidth?: number;
+    $height?: number;
+    $minHeight?: number;
+    $maxHeight?: number;
+    x?: number; // TODO not really used
+    y?: number;
+};
+
+const Resizers = styled.div<ResizersProps>(({ $width, $minWidth, $maxWidth, $height, $minHeight, $maxHeight, x, y }) => `
+        ${$width ? `width: ${$width}px;` : 'width: auto;'};
+        ${$minWidth && `min-width: ${$minWidth}px;`};
+        ${$maxWidth && `max-width: ${$maxWidth}px;`};
+        ${$height ? `height: ${$height}px;` : 'height: auto;'};
+        ${$minHeight && `min-height: ${$minHeight}px;`};
+        ${$maxHeight && `max-height: ${$maxHeight}px;`};
+        // ${x && `left: ${x}px;`};
+        // ${y && `top: ${y}px;`};
+        box-sizing: border-box;
+        position: relative;
+
+        #top-div, #left-div, #right-div, #bottom-div {
+          position: absolute;
+        }
+
+        #top-div, #bottom-div {
+          width: 100%;
+          height: ${reactiveAreaWidth};
+          cursor: ns-resize;
+        }
+
+        #left-div, #right-div {
+          width: ${reactiveAreaWidth};
+          height: 100%;
+          bottom: 0;
+          cursor: ew-resize;
+        }
+
+        #top-div {
+          top: ${`-${reactiveAreaWidth}`};
+        }
+
+        #left-div {
+          left: ${`-${reactiveAreaWidth}`};
+        }
+
+        #right-div {
+          right: ${`-${reactiveAreaWidth}`};
+        }
+
+        #bottom-div {
+          bottom: ${`-${reactiveAreaWidth}`};
```

Comment:
This depends on our decision. I was checking how this looks in other apps, and mostly the interactive area was outside the box, not in the center of the border line, most of the times.. Probably because this way one could click on/inside the box and focus on it without triggering any resizing actions.